homework10/FileLogger.java

```diff
@@ -0,0 +1,57 @@
+package homework10;
+
+import homework10.configuration.FileLoggerConfiguration;
+import homework10.enums.LoggingLevel;
+import homework10.exceptions.FileMaxSizeReachedException;
+import homework10.interfaces.Printable;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.Random;
+
+public class FileLogger implements Printable {
+
+    private final FileLoggerConfiguration config;
+
+    public FileLogger(FileLoggerConfiguration config) {
+        this.config = config;
+    }
+
+
+    public void info(String message){
+        writeInFile(message, LoggingLevel.INFO);
+    }
+
+
+    public void debug(String message){
+        writeInFile(message, LoggingLevel.DEBUG);
+    }
+
+    @Override
+    public void writeInFile(String message, LoggingLevel levels) {
+        File file = new File(config.getFilename());
+        Date timestamp = new Date();
+
+        if (file.exists() && file.length() > config.getMaxSize()) {

```

Comment:
This `file.length() > config.getMaxSize()` can be extracted to a separate method `isMaxFileSizeExceeded`